api/preprint_providers/views.py

```diff
@@ -0,0 +1,189 @@
+
+from api.preprint_providers.serializers import PreprintProviderSerializer
+from api.preprints.serializers import PreprintSerializer
+
+
+class PreprintProviderMixin(object):
+    institution_lookup_url_kwarg = 'institution_id'
+
+    def get_preprintprovider(self):
+        provider = get_object_or_error(
+            PreprintProvider,
+            self.kwargs[self.institution_lookup_url_kwarg],
+            display_name='institution'
+        )
+        return provider
+
+
+class PreprintProviderList(JSONAPIBaseView, generics.ListAPIView, ODMFilterMixin):
+    """
+    Paginated list of verified PreprintProviders available
+
+    ##PreprintProvider Attributes
+
+    OSF Institutions have the "institutions" `type`.
+
+        name           type               description
+        =========================================================================
+        name           string             title of the preprint provider
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the preprint provider's static logo
+        banner_path    string             a path to the preprint provider's static banner
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.INSTITUTION_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model_class = PreprintProvider
+
+    pagination_class = MaxSizePagination
+    serializer_class = PreprintProviderSerializer
+    view_category = 'preprint_providers'
+    view_name = 'preprint_providers-list'
+
+    ordering = ('name', )
+
+    def get_default_odm_query(self):
+        return Q('_id', 'ne', None)
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return PreprintProvider.find(self.get_query_from_request())
+
+
+class PreprintProviderDetail(JSONAPIBaseView, generics.RetrieveAPIView, PreprintProviderMixin):
+    """ Details about a given preprint provider.
+
+    ##Attributes
+
+    Preprint Providers have the "preprint_providers" `type`.
+
+        name           type               description
+        =========================================================================
+        name           string             title of the preprint provider
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the preprint provider's static logo
+        banner_path    string             a path to the preprint provider's static banner
+
+    ##Links
+
+        self:  the canonical api endpoint of this institution
+        html:  this institution's page on the OSF website
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ]
```

Comment:
See above edited https://github.com/CenterForOpenScience/osf.io/pull/6166#discussion_r75343106

This should be `ALWAYS_PUBLIC`